public/pages/workflow_detail/workflow_inputs/processors_list.tsx

```diff
@@ -36,13 +39,20 @@ import {
   SplitIngestProcessor,
   SplitSearchResponseProcessor,
   TextChunkingIngestProcessor,
+  TextEmbeddingIngestProcessor,
+  TextImageEmbeddingIngestProcessor,
 } from '../../../configs';
 import { ProcessorInputs } from './processor_inputs';
+import { SavedObject } from '../../../../../../src/core/public';
+import { DataSourceAttributes } from '../../../../../../src/plugins/data_source/common/data_sources';
+import { useLocation } from 'react-router-dom';
 
 interface ProcessorsListProps {
   uiConfig: WorkflowConfig;
   setUiConfig: (uiConfig: WorkflowConfig) => void;
   context: PROCESSOR_CONTEXT;
+  dataSource?: SavedObject<DataSourceAttributes>;
```

Comment:
correct me if I'm wrong, I believe we can remove this added `dataSource` prop now. I see you've added the helper fn to get the data source details (including the version) based on the data source ID - `getEffectiveVersion()`.

Suggest to remove the added props here and in any parent components, wherever it's not needed. That should help minimize the changes across many of the changed files.

Reply:
addressed on line 53.